### PR TITLE
Docs and guides for `--use-wallet`

### PR DESCRIPTION
**Pull request by peterwht**
Direct links:
- [call contract](https://github.com/r0gue-io/pop-docs/blob/peter/docs-use-wallet/pop-cli-for-smart-contracts/pop-cli/call.md)
- [up contract](https://github.com/r0gue-io/pop-docs/blob/peter/docs-use-wallet/pop-cli-for-smart-contracts/pop-cli/up.md)
- [Guide: contract](https://github.com/r0gue-io/pop-docs/blob/peter/docs-use-wallet/pop-cli-for-smart-contracts/guides/securely-sign-transactions-from-cli.md)
- [call chain](https://github.com/r0gue-io/pop-docs/blob/peter/docs-use-wallet/pop-cli-for-appchains/pop-cli/call.md)
- [Guide: chain](https://github.com/r0gue-io/pop-docs/blob/peter/docs-use-wallet/pop-cli-for-appchains/guides/securely-sign-transactions-from-cli.md)

### CODE REVIEW

**Comment by peterwht** [comment]
[sc-1779]

File: pop-cli-for-smart-contracts/.gitbook/assets/use-wallet/connect-wallet.png
Comment (al3mart) [inline pop-cli-for-smart-contracts/.gitbook/assets/use-wallet/connect-wallet.png:1]: We could substitute with a screen shot showing many more wallets!

Gave a heads up to @AlexD10S as he already has this.

File: pop-cli-for-smart-contracts/guides/securely-sign-transactions-from-cli.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Securely Sign Transactions from CLI
+Pop CLI provides an option to securely sign transactions from the CLI. It does this by opening a signing portal allowing you
+to sign the transactions using your browser extension wallet.
+
+## Example Usage: Deploying a Contract
+For a full guide on deploying a contract, see the [Deploy Your Contract Locally](./deploy-your-contract-locally.md) guide.
+
+Normally, you would provide `--suri=<private-key>` to deploy the contract. However, this should only be used for development accounts.
```
Comment (al3mart) [inline pop-cli-for-smart-contracts/guides/securely-sign-transactions-from-cli.md:8]: I feel we should append something at the end here mentioning that this implies insecure handling of secretes.

Comment (al3mart) [reply pop-cli-for-smart-contracts/guides/securely-sign-transactions-from-cli.md:8]: https://github.com/r0gue-io/pop-docs/pull/17

File: pop-cli-for-smart-contracts/.gitbook/assets/use-wallet/connect-wallet.png
Comment (AlexD10S) [reply pop-cli-for-smart-contracts/.gitbook/assets/use-wallet/connect-wallet.png:1]: Captura de pantalla 2024-12-18 a las 12 49 03

**Review by AlexD10S — COMMENTED** [review]
Looks good! 
I included a brief comment about wallet integration in the parachain onboarding docs.

Also as suggested by Ale here: https://github.com/r0gue-io/pop-docs/pull/16#discussion_r1889969779 I updated image to show all wallets, but feel free to pick the one you prefer.

**Review by AlexD10S — APPROVED** [review]
Looks good to me!

File: pop-cli-for-smart-contracts/guides/securely-sign-transactions-from-cli.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+```
+
+Your browser will open a new tab with the following screen at http://127.0.0.1:9090 (or similar if the port is already in use);
+<figure><img src="../.gitbook/assets/use-wallet/unconnected-wallet.png" alt="" width="450"><figcaption><p>Signing Portal Initial Open</p></figcaption></figure>
+
+Click on the `Connect Wallet` button to connect your browser extension wallet.
+<figure><img src="../.gitbook/assets/use-wallet/connect-wallet.png" alt="" width="450"><figcaption><p>Connect Wallet</p></figcaption></figure>
+
+After connecting your wallet, you will have the option to choose your account and sign the transaction, and finally see the transaction details for signing.
+Note that for contract calls, the portal will perform a dry-run to alert if the transaction will succeed, and also estimate the gas costs.
```
Comment (evilrobot-01) [inline pop-cli-for-smart-contracts/guides/securely-sign-transactions-from-cli.md:30]:
```suggestion
Note that for contract calls, the portal will perform a dry-run to alert you if the transaction will succeed, and also estimate the gas costs.
```

File: pop-cli-for-smart-contracts/guides/securely-sign-transactions-from-cli.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+<figure><img src="../.gitbook/assets/use-wallet/connected-wallet.png" alt="" width="450"><figcaption><p>Transaction Details</p></figcaption></figure>
+
+If the transaction will fail, the dry-run alert will change:
+<figure><img src="../.gitbook/assets/use-wallet/fail-dry-run.png" alt="" width="450"><figcaption><p>Failing Dry Run</p></figcaption></figure>
+
+In this case, you can expand the dry-run alert to see the error details. This example shows an error of the account not having enough funds:
+<figure><img src="../.gitbook/assets/use-wallet/expanded-dry-run-fail.png" alt="" width="450"><figcaption><p>Expanded Failing Dry Run</p></figcaption></figure>
+
+Once ready to sign, pressing the `Sign` button will open your wallet for signature.
```
Comment (evilrobot-01) [inline pop-cli-for-smart-contracts/guides/securely-sign-transactions-from-cli.md:39]:
```suggestion
Once ready to sign, pressing the `Submit` button will open your wallet for signature.
```

File: pop-cli-for-appchains/guides/securely-sign-transactions-from-cli.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+Your browser will open a new tab with the following screen at http://127.0.0.1:9090 (or similar if the port is already in use);
+<figure><img src="../../pop-cli-for-smart-contracts/.gitbook/assets/use-wallet/unconnected-wallet.png" alt="" width="450"><figcaption><p>Signing Portal Initial Open</p></figcaption></figure>
+
+Click on the `Connect Wallet` button to connect your browser extension wallet.
+<figure><img src="../../pop-cli-for-smart-contracts/.gitbook/assets/use-wallet/connect-wallet.png" alt="" width="450"><figcaption><p>Connect Wallet</p></figcaption></figure>
+
+After connecting your wallet, you will have the option to choose your account and sign the transaction, and finally see the transaction details for signing.
+<figure><img src="../.gitbook/assets/use-wallet/connected-wallet.png" alt="" width="450"><figcaption><p>Transaction Details</p></figcaption></figure>
+
+Once ready to sign, pressing the `Sign` button will open your wallet for signature.
```
Comment (evilrobot-01) [inline pop-cli-for-appchains/guides/securely-sign-transactions-from-cli.md:32]:
```suggestion
Once ready to sign, pressing the `Submit` button will open your wallet for signature.
```

**Review by evilrobot-01 — CHANGES_REQUESTED** [review]
Just three quick fixes please and then good to approve.

**Review by evilrobot-01 — APPROVED** [review]
Looks good, thank you!

**Pull request merged**